scripts/west_commands/runners/probe_rs.py

```diff
@@ -88,7 +89,15 @@ def do_flash(self, **kwargs):
         download_args = []
         if self.erase:
             download_args += ['--chip-erase']
-        download_args += [self.elf_name]
+
+        if self.hex_file is not None:
+            download_args += ['--binary-format', 'hex']
+            download_args += [self.hex_file]
+        else:
+            if self.elf_name is None:
+                raise ValueError(f'Cannot flash; elf file required.')
```

Comment:
Sorry to disagree with @pdgendt but the nesting does not seem needed. This is also has more consistent negations and looks more like a case/switch (which could also be an option?

```python
  if self.hex_file is not None:
     ...
  elif self.elf_name is not None:
     ...
  else:
    raise ValueError(f'Cannot flash; elf file required.')
```

No big deal.